tests/src/buildApplication/generateFunctionsMap.test.ts

```diff
@@ -50,11 +90,33 @@ describe('generateFunctionsMap', async () => {
 			'validTest/functions',
 			false
 		);
-		expect(invalidFunctions.size).toEqual(0);
-		expect(functionsMap.size).toEqual(3);
-		expect(functionsMap.get('index')).toMatch(/\/index\.func\.js$/);
-		expect(functionsMap.get('index.rsc')).toMatch(/\/index\.rsc\.func\.js$/);
-		expect(functionsMap.get('api/hello')).toMatch(/\/api\/hello\.func\.js$/);
+
+		expect(invalidFunctions.size).toEqual(1);
```

Comment:
I know. Sorry about that. I was trying not to completely change your function mock implementation while extending the tests to support the new behaviour where it squashes route groups. So, I am testing that invalid functions at the base do not work, while a valid squashed function replaces an invalid function at the base.

I know, it's not nice in the current implementation. Changing how we mock the file system will vastly help and improve these tests. I just wanted to try and test the additional edge cases without changing much about how you were mocking it.

Reply:
ah I see, ok thanks, could you please change the test description accordingly and/or add a comment here then? (just so that we can later split this when we get better fs mocking 🙂)